### PR TITLE
Improved entry point of the documentation.

### PR DESCRIPTION
**Pull request by frisso**
Documentation updated with more details, better explanation, and well-formed.

### CODE REVIEW

File: Documentation/intro.rst [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+``Polycube`` is an **open source** software framework that provides **fast** and **lightweight** **network functions**. Already available services, called `cubes`, include `bridge`, `router`, `nat`, `load balancer`, `firewall`, `DDoS mitigator`, and more.
+
+Cube can be composed to build arbitrary **service chains** and provide custom network connectivity to **namespaces**, **containers**, **virtual machines**, and **physical hosts**.
+
+Cubes are extremely **efficient** because are based on the recent `BPF` and `XDP` Linux kernel technologies. In addition, cubes are easily **extensible** and **customizable**.
+
+Polycube networks can be controlled by a coherent, uniform and simple command line, named `polycubectl`, which interacts with a REST-based daemon, `polycubed`.
+
+Polycube provides also two proof-of-concept **standalone applications** based upon this framework.
```
Comment (mbertrone) [inline Documentation/intro.rst]: `proof-of-concept` -> `proofs-of-concept`

File: Documentation/intro.rst [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Polycube greatly simplifies the way data, control, and management planes communicate, hence enabling the creation of rich network services that include all the above features.
+In turn, this offers a simplified environment to service developers, who can leverage the power of Polycube to write the data/control and management parts of their network services.
+
+In detail, most network services include a *data plane*, such as the longest prefix match algorithm in a router, a *control plane*, e.g., where routing protocols are executed, and a *management plane*, devoted to the configuration and monitoring of the service.
+`bpf` aims at the creation of fast data planes, leaving the rest under the responsibility of the developer; Polycube overcomes this limitation with a rich set of primitives natively provided by the framework.
+In addition, Polycube provides the software infrastructure required to overcome possible limitations of BPF in the data plane (e.g., the limited size of data plane programs), enabling the steering of packets that require complex processing in user-space, where previous limitations doesn't apply.
```
Comment (mbertrone) [inline Documentation/intro.rst]: limitations `doesn't` -> `don't` apply

File: Documentation/intro.rst [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+Cube can be composed to build arbitrary **service chains** and provide custom network connectivity to **namespaces**, **containers**, **virtual machines**, and **physical hosts**.
+
+Cubes are extremely **efficient** because are based on the recent `BPF` and `XDP` Linux kernel technologies. In addition, cubes are easily **extensible** and **customizable**.
+
+Polycube networks can be controlled by a coherent, uniform and simple command line, named `polycubectl`, which interacts with a REST-based daemon, `polycubed`.
+
+Polycube provides also two proof-of-concept **standalone applications** based upon this framework.
+`Pcn-k8s` is a brand new network plug-in for *Kubernetes*, which can handle the network of an entire datacenter and that outperforms existing solutions in terms of throughput.
+`Pcn-iptables` is a more efficient and scalable clone of *iptables*.
```
Comment (acloudiator) [inline Documentation/intro.rst]:
```suggestion
`PCN-iptables` is a more efficient and scalable clone of existing Linux *iptables*.
```

File: Documentation/intro.rst [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+Main features
+-------------
+
+Extremely fast
+~~~~~~~~~~~~~~
+Polycube enables extremely fast and efficient network services, thanks to its capability to run inside the Linux kernel and, whenever possible, as close as possible to the network interface card driver, which reduces the time spent in ancillary components. 
+
+
+Born for service chaining
```
Comment (acloudiator) [inline Documentation/intro.rst]: Change of word for `Born`

File: Documentation/intro.rst [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+Polycube supports the definition of multiple network scenarios through the composition of many elementary building blocks (i.e., *cubes*),  which can be combined (e.g., attached one to the other) to create complex network services.
+For instance, several dockers can communicate through a bridge, which is then attached to a router to provide internet connectivity (possibly through a nat), while a firewall protects the entire infrastructure.
+
+Polycube has been designed to simplify service chaining: cubes can be dynamically instantiated and seamlessly connected together using virtual links, mimicking traditional networks in which dedicated middlebox are connected with each other through physical wires.
+As a consequence, cubes can be composed to build arbitrary service chains and provide custom network connectivity to namespaces, docker, virtual machines, and physical hosts.
+
+
+Production-grade network services
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Polycube greatly simplifies the way data, control, and management planes communicate, hence enabling the creation of rich network services that include all the above features.
```
Comment (acloudiator) [inline Documentation/intro.rst]:
```suggestion
Polycube greatly simplifies the data, control, and management planes communication, hence it enables the creation of rich network services which includes all the above features.
```

File: Documentation/intro.rst (hunk 2 above)
Comment (acloudiator) [reply Documentation/intro.rst]: It should be fine, third-person speech used here.

**Comment by acloudiator** [comment]
Very well documented, just a few small notes.

File: Documentation/intro.rst (hunk 1 above)
Comment (frisso) [reply Documentation/intro.rst]: Not sure it's correct. The plural refers to applications. Let's wait some native speaker to take care of this.

File: Documentation/components/iptables/pcn-iptables.rst [hunk 6]
```diff
@@ -1,9 +1,12 @@
-pcn-iptables: An iptables clone based on eBPF
-=============================================
+pcn-iptables: a clone of iptables based on eBPF
+===============================================
+
+Polycube includes the ``pcn-iptables`` standalone application, a stateful firewall whose syntax and semantic is compatible with the well-known ``iptables`` Linux tool.
```
[the remark this answers is not in the document]
Comment (frisso) [reply Documentation/components/iptables/pcn-iptables.rst]: Let's keep stateful firewall. The application is a firewall, not a ruleset.
Ok for the second comment.

File: Documentation/intro.rst (hunk 3 above)
Comment (frisso) [reply Documentation/intro.rst]: Used 'lowercase' in the rest of the text; changed to 'pcn-iptables'.

**Comment by frisso** [comment]
Except the commit message that is not that good, I should have addressed all your comments.

**Comment by frisso** [comment]
@acloudiator Are you ok? Can we merge?

**Pull request merged**